src/dataframe/dataframe.jl

```diff
@@ -466,38 +466,65 @@ function Base.setindex!(df::DataFrame, v::Any, row_ind::Real, col_ind::ColumnInd
     insert_single_entry!(df, v, row_ind, col_ind)
 end
 
-# df[SingleRowIndex, MultiColumnIndex] = Single Item
+# df[SingleRowIndex, :] = Union{AbstractDict, NamedTuple}
 function Base.setindex!(df::DataFrame,
-                        v::Any,
-                        row_ind::Real,
-                        col_inds::AbstractVector{Bool})
-    setindex!(df, v, row_ind, findall(col_inds))
-end
-function Base.setindex!(df::DataFrame,
-                        v::Any,
+                        row::Union{AbstractDict, NamedTuple},
                         row_ind::Real,
-                        col_inds::AbstractVector{<:ColumnIndex})
-    for col_ind in col_inds
-        insert_single_entry!(df, v, row_ind, col_ind)
+                        ::Colon)
+    old_row = df[row_ind, :]
+    for nm in _names(df)
+        try
+            val = row[nm]
+            insert_single_entry!(df, val, row_ind, nm)
+        catch
+            # clean up partial row
+            df[row_ind, :] = old_row
+            msg = "Error adding value to column :$nm."
+            throw(ArgumentError(msg))
+        end
     end
-    return df
+    df
 end
 
-# df[SingleRowIndex, MultiColumnIndex] = 1-Row DataFrame
+# df[SingleRowIndex, MultiColumnIndex] = iterable (or scalar, being deprecated)
 function Base.setindex!(df::DataFrame,
-                        new_df::DataFrame,
+                        v::Any,
                         row_ind::Real,
                         col_inds::AbstractVector{Bool})
-    setindex!(df, new_df, row_ind, findall(col_inds))
+    setindex!(df, v, row_ind, findall(col_inds))
 end
 function Base.setindex!(df::DataFrame,
-                        new_df::DataFrame,
+                        v::Any,
                         row_ind::Real,
                         col_inds::AbstractVector{<:ColumnIndex})
```

Comment:
We should handle the case when `col_inds` have duplicates also `AbstractVector{<:ColumnIndex}` allows mixing numbers and symbols which should not be allowed.